internal/sign/job/manager.go

```diff
@@ -5,40 +5,102 @@ import (
 	"errors"
 	"fmt"
 
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
 	kmmv1beta1 "github.com/kubernetes-sigs/kernel-module-management/api/v1beta1"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/module"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/registry"
 	"github.com/kubernetes-sigs/kernel-module-management/internal/sign"
 	"github.com/kubernetes-sigs/kernel-module-management/internal/utils"
-	"sigs.k8s.io/controller-runtime/pkg/log"
 )
 
 type signJobManager struct {
+	client    client.Client
 	signer    Signer
 	helper    sign.Helper
 	jobHelper utils.JobHelper
+	registry  registry.Registry
 }
 
-func NewSignJobManager(signer Signer, helper sign.Helper, jobHelper utils.JobHelper) *signJobManager {
+func NewSignJobManager(
+	client client.Client,
+	signer Signer,
+	helper sign.Helper,
+	jobHelper utils.JobHelper,
+	registry registry.Registry) *signJobManager {
 	return &signJobManager{
+		client:    client,
 		signer:    signer,
 		helper:    helper,
 		jobHelper: jobHelper,
+		registry:  registry,
 	}
 }
 
-func (jbm *signJobManager) Sync(ctx context.Context, mod kmmv1beta1.Module, m kmmv1beta1.KernelMapping, targetKernel string, imageToSign string, targetImage string, pushImage bool) (utils.Result, error) {
+func (jbm *signJobManager) ShouldSync(
+	ctx context.Context,
+	mod kmmv1beta1.Module,
+	m kmmv1beta1.KernelMapping,
+	opts ...sign.SyncOption) (bool, error) {
+
+	options := sign.SyncOptions{}
+	for _, opt := range opts {
+		opt(&options)
+	}
+	if options.Owner == nil {
+		options.Owner = &mod
+	}
+
+	// if there is no sign specified skip
+	if !module.ShouldBeSigned(mod.Spec, m) {
+		return false, nil
+	}
+
+	exists, err := module.ImageExists(ctx, jbm.client, jbm.registry, mod.Spec, options.Owner.GetNamespace(), m, m.ContainerImage)
+	if err != nil {
+		return false, fmt.Errorf("failed to check existence of image %s: %w", m.ContainerImage, err)
+	}
+
+	return !exists, nil
+}
+
+func (jbm *signJobManager) Sync(
+	ctx context.Context,
+	mod kmmv1beta1.Module,
+	m kmmv1beta1.KernelMapping,
+	targetKernel string,
+	opts ...sign.SyncOption) (utils.Result, error) {
+
+	options := sign.SyncOptions{}
+	for _, opt := range opts {
+		opt(&options)
+	}
+	if options.Owner == nil {
+		options.Owner = &mod
+	}
+
 	logger := log.FromContext(ctx)
 
 	logger.Info("Signing in-cluster")
 
-	signConfig := jbm.helper.GetRelevantSign(mod, m)
-	jobLabels := jbm.jobHelper.JobLabels(mod, targetKernel, "sign")
+	signConfig := jbm.helper.GetRelevantSign(mod.Spec, m)
```

Comment:
The same comment as in Build: WDYT regarding moving all the jobOptions into the MakeJobTemplate?